src/uct/ib/mlx5/ece.h

```diff
@@ -0,0 +1,79 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCT_ECE_H_
+#define UCT_ECE_H_
+
+
+#include <uct/ib/base/ib_log.h>
+#include <uct/ib/base/ib_device.h>
+#include <ucs/debug/log.h>
+#include <ucs/type/status.h>
+
+#define UCT_IB_MLX5_VENDOR_ID             0x15b3
+#define UCT_IB_MLX5_ECE_VER_MAX           2
+#define UCT_IB_MLX5_ECE_VER_SR            2
+#define UCT_IB_MLX5_ECE_VER_SHIFT         28
+#define UCT_IB_MLX5_ECE_SR                0x1
+#define UCT_IB_MLX5_ECE_UNKNOWN_MASK      0x0ffffffe
```

Comment:
can be created by macro
UCS_MASK(UCT_IB_MLX5_ECE_VER_SHIFT) ^ UCT_IB_MLX5_ECE_FLAG_SELECTIVE_REPEAT